src/auth.js

```diff
@@ -142,22 +142,41 @@ export default class DropboxAuth {
   }
 
   generatePKCECodes() {
-    let codeVerifier = crypto.randomBytes(PKCELength);
-    codeVerifier = codeVerifier.toString('base64')
-      .replace(/\+/g, '-')
-      .replace(/\//g, '_')
-      .replace(/=/g, '')
-      .substr(0, 128);
+    let codeVerifier;
+    function convertToStringAndReplace(toBeConverted) {
```

Comment:
This can be a util, also naming this something more appropriate will help keep it maintainable

Reply:
Also, why is substring not included in this function?

Reply:
yea I can abstract it. I didn't include substr because we essentially do the same logic later just without substr so I just reused the func